### PR TITLE
Added trailer to TV Shows

### PR DESCRIPTION
**Pull request by tielis**
Add TV Show trailer

### CODE REVIEW

File: libs/data_utils.py [hunk 1]
```diff
@@ -27,6 +27,13 @@
 from .utils import safe_get, logger
 from . import settings
 
+try: #PY2 / PY3
```
Comment (pkscout) [inline libs/data_utils.py]: We are maintaining separate Leia and Matrix branches, so for the Matrix PR I would want only the needed Python3 imports.

File: libs/data_utils.py [hunk 2]
```diff
@@ -347,3 +357,26 @@ def parse_media_id(title):
...
+                key = result.get('key')
+                chk_link = "https://www.youtube.com/oembed?format=json&url=https://www.youtube.com/watch?v="+key            
+                try:
+                    check = urlopen(chk_link)
+                except URLError as e:
+                    erro = e.code
+                if erro == 404:                        # video NOT available 
+                    pass
+                elif result.get('type') == 'Trailer':  # video is available and is defined as "Trailer" by TMDB. Perfect link!           
+                    return 'plugin://plugin.video.youtube/?action=play_video&videoid='+key
```
Comment (pkscout) [inline libs/data_utils.py]: I don't like the fact that this requires the YouTube addon given that there is a newer (and much easier to use) Tubed addon (at least for Matrix).  There might need to be a setting to select which YouTube addon to use.

File: libs/data_utils.py [hunk 3]
```diff
@@ -347,3 +357,26 @@ def parse_media_id(title):
...
+def _parse_trailer(results):
+    if results:        
+        keyBackup = None
+        for result in results:
+            if result.get('site') == 'YouTube':
+                erro = None
+                key = result.get('key')
+                chk_link = "https://www.youtube.com/oembed?format=json&url=https://www.youtube.com/watch?v="+key            
+                try:
+                    check = urlopen(chk_link)
```
Comment (pkscout) [inline libs/data_utils.py:380]: I really dislike the idea of waiting for a URL check for every TV show, especially if the user doesn't want the trailers or isn't using a skin that supports them.  I might be OK with this if there was a setting to enable trailer downloads that defaults to `False`.

**Comment by pkscout** [comment]
As with the other PR, if this is accepted at all (and I'm still pondering), I would want a separate PR for the Leia branch as well.

**Comment by KarellenX** [comment]
I know that Trailers are a missing feature for TV Shows. Local trailers are not scanned and they need to be forced in a clunky way.

There has never been a tv show scraper with trailer features, so I cannot confirm whether Kodi will treat this correctly.

Have you tested this in Kodi @tielis ?

Thanks for the contributions :)

**Comment by pkscout** [comment]
Thanks for your thoughts @KarellenX.  I do like the idea having it as an optional setting in the scraper.

**Comment by KarellenX** [comment]
Yes, I prefer the optional setting also.

There  used to be a bug where if you had local trailers but had the Trailer setting in the scraper enabled, the local trailer would be ignored in preference for the online link which is contrary to how Kodi treats all other local information. For this reason  I used to always disable the trailer setting in the scraper.

But I cannot seem to replicate that behaviour in v19 anymore, nor could I find any fix related to this issue. Probably safer to have the setting in case the bug reappears.

**Comment by KarellenX** [comment]
Just had a quick search as I recalled discussing trailers for tv shows once before with AkariDN. 

https://github.com/xbmc/xbmc/pull/13566#issuecomment-495976133

**Comment by tielis** [comment]
@KarellenX  I have been using kodi with this change for a while, but I haven't tested it with local Trailers, so I don't know how it will behave.
I'll be implementing it as optimal setting then.

File: libs/data_utils.py (hunk 2 above)
Comment (tielis) [reply libs/data_utils.py]: What is the url command to play youtube videos in this Tubed addon?

Comment (tielis) [reply libs/data_utils.py]: don't worry, I've got it here.

plugin://plugin.video.tubed/?mode=play&video_id=

**Comment by tielis** [comment]
Do we really need to put Trailer option default off ? People are so used to not have trailers for TV Shows that no one will notice we have it available now. I would vote to leave it as default 'on' and if bugs start to come up, we change to off. Trailers for movies are default on, Tv shows should be the same. 

As soon you give a thumbs up, I will PR for Leia too.

File: resources/settings.xml [hunk 4]
```diff
@@ -146,6 +146,25 @@
 					<default>true</default>
 					<control type="toggle"/>
 				</setting>
+        <setting id="enab_trailer" type="boolean" label="30305" help="">
```
Comment (pkscout) [inline resources/settings.xml:149]: The fact that the movie scraper has it enabled by default is a compelling argument, so I'm fine leaving it defaulted to `True` for now. It's easy enough to change before we push it into production if we decide to leave it disabled by default. My main concern is the amount of time it adds to the scrape, so I'll do some testing when I'm back in a place where I have access to my full library.

File: resources/settings.xml [hunk 5]
```diff
@@ -146,6 +146,25 @@
...
+          <level>0</level>
+          <default>true</default>
+          <control type="toggle"/>
+        </setting>
+        <setting id="players_opt" type="string" parent="enab_trailer" label="30306" help="">
+          <level>0</level>
+          <default>Tubed</default>
```
Comment (pkscout) [inline resources/settings.xml:156]: I'd like the strings here included in the strings file. I know it's the name of the addons, but it is possible those names might be different in different languages.

**Comment by pkscout** [comment]
Other than the strings comment, this looks good.  With that last change, I'm willing to merge this and then do the same when you get the Leia one done. Thanks for your contribution.

**Review by pkscout — APPROVED** [review]

**Pull request merged**